test/e2e/framework/util.go

```diff
@@ -1284,3 +1285,110 @@ func taintExists(taints []v1.Taint, taintToFind *v1.Taint) bool {
 	}
 	return false
 }
+
+// WatchEventSequenceVerifier ...
+// manages a watch for a given resource, ensures that events take place in a given order, retries the test on failure
+//   testContext         cancelation signal across API boundries, e.g: context.TODO()
+//   dc                  sets up a client to the API
+//   resourceType        specify the type of resource
+//   namespace           select a namespace
+//   resourceName        the name of the given resource
+//   listOptions         options used to find the resource, recommended to use listOptions.labelSelector
+//   expectedWatchEvents array of events which are expected to occur
+//   scenario            the function to run
+func WatchEventSequenceVerifier(testContext context.Context, dc dynamic.Interface, resourceType schema.GroupVersionResource, namespace string, resourceName string, listOptions metav1.ListOptions, expectedWatchEvents []watch.Event, scenario func(*watchtools.RetryWatcher) []watch.Event) {
```

Comment:
why `testContext` instead of `context`?

Reply:
`expectedWatchEvents []watch.Event` seems too permissive:
- we're not comparing events, just their types
- we're not comparing event types, but their order

Reply:
> why `testContext` instead of `context`?

This was used because `context` would be the same name as the import, seems to be fine with the same name though

Reply:
`ctx` is typical

Reply:
I'll update it to be `ctx`, so it's not the same